extension/src/background/index.ts

```diff
@@ -8,15 +9,37 @@ async function polling() {
     log(`Polling`);
 }
 
-async function fetchPreviousRoomId() {
+async function fetchPreviousRoomId(): Promise<{ [s: string]: any }> {
     // Maybe boot a random roomId on first load ?
     return browser.storage.sync.get({ roomId: undefined });
 }
-// TODO : Connect to previous roomId using websocket
+
+const wsClient: WebSocketClient = new WebSocketClient(process.env.WS_URL || "");
+
+async function connectToWebSocket() {
+    try {
+        await wsClient.connect();
+        console.log("Connected to WebSocket");
+        wsClient.getWebSocket().onmessage = (event: any) => {
+            // Since the messages are supposed to be in JSON format, should be: JSON.parse(event.data)
+            const receivedMsg = event.data;
+            console.log(receivedMsg);
+        };
+    } catch (e) {
+        console.log(e);
+    }
+}
+
+// Connect to previous roomId using websocket
+async function connectToPreviousRoom() {
+    const roomId = await fetchPreviousRoomId();
+    wsClient.getWebSocket().send(JSON.stringify({ roomId }));
```

Comment:
I think we should start making types/enums for message communications. Something that would resemble what we're doing with backeng/content/popup script communications (and why not use the same enum ?).
 ```suggestion
    wsClient.getWebSocket().send(JSON.stringify({ type: <enum>.JOIN_ROOM, roomId }));
```

regarding the name of <enum> here are a few suggestions
- Reuse the existing `MessageType` (this enum can be used both for internal communication between scripts AND websocket stuff
- `WebsocketMessageType` (and rename `MessageType` to `ExtensionMessageType`)

I think it could be a good idea to reuse the existing MessageType enum at first, and see if later we need to split them. 

Also, maybe instead of JOIN_ROOM we can use the existing event CHANGE_ROOM. After all, if a user joins a new room this also means, that in our DB, we'll have to clear the previous connection if it remained and wasn't properly disconnected

Reply:
EDIT : we need to use a key as a routing identifier to pick the AWS lambda function that will be executed... since it's maybe a bit out of scope of your PR, then you can leave it as is and I will handle the rest

Reply:
I used the enum `MessageType.CHANGE_ROOM`, to stay consistent with the `case` block.
And yes, in the future, we will need to send/receive more message types, with a different structure, so we'll have to look at how these keys work